src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,108 @@
+package duke;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Scanner;
+
+public class Parser {
+    enum Commands {
+        LIST("list"),
+        MARK("mark"),
+        UNMARK("unmark"),
+        DELETE("delete"),
+        TODO("todo"),
+        DEADLINE("deadline"),
+        EVENT("event"),
+        BYE("bye");
+
+
+        final String command;
+
+        Commands(String command) {
+            this.command = command;
+        }
+    }
+
+    protected static void start() throws DukeException, IOException {
+        //to read user input
+        Scanner scanned = new Scanner(System.in);
+        System.out.println("");
+        String input = scanned.nextLine();
+
+        while (!input.equals(Commands.BYE.command)) {
+            System.out.println("=======================================================================");
+
+            //prints the list of items when user inputs list
+            if (input.equals(Commands.LIST.command)) {
+                printList(Duke.list);
+            }
+
```

Comment:
It might be better to follow the standard if-else format. Perhaps something like:
} else if (input.length() > 4 && Commands.MARK.command.equals(input.substring(0, 4))) {
                //marks a task as done if the user inputs a string in format "mark xx"
               //NOTE: currently, inputting "mark" alone results in a task named "mark" to be added to the list of tasks
                TaskList.mark(input);
}

Reply:
thank you for the catch! I will update it soon! :)